domain/machine/state/machine_cloud_instance_test.go

```diff
@@ -0,0 +1,194 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package state
+
+import (
+	"context"
+
+	jc "github.com/juju/testing/checkers"
+	gc "gopkg.in/check.v1"
+
+	"github.com/juju/juju/core/instance"
+)
+
+func (s *stateSuite) TestGetHardwareCharacteristics(c *gc.C) {
+	db := s.DB()
+
+	// Create a reference machine.
+	err := s.state.UpsertMachine(context.Background(), "42")
+	c.Assert(err, jc.ErrorIsNil)
+	var machineUUID string
+	row := db.QueryRowContext(context.Background(), "SELECT uuid FROM machine WHERE machine_id=\"42\"")
```

Comment:
SQL uses single quotes for strings. You then don't need to escape all the doubles.

Reply:
Fixed.